src/adapter/src/coord.rs

```diff
@@ -6462,12 +6467,42 @@ pub mod fast_path_peek {
         }));
     }
 
+    impl<T> Plan<mz_expr::OptimizedMirRelationExpr, T> {
+        pub fn finalize<F>(
```

Comment:
Would it be possible to make `create_plan` do the work of this function? If I'm not mistaken, the only actual thing happening here is the call to `finalize_dataflow`, so maybe that call could be added to `create_plan`. And then `Plan` and `PeekDataflowPlan` wouldn't have to be made generic in the IR.

Reply:
It technically can, but the point of making this two step is so that we can print out fast path information as part of both `EXPLAIN OPTIMIZED PLAN` and `EXPLAIN PHYSICAL PLAN`.

Reply:
Ah, ok!